.vscode/scripts/create-simulation.sh

```
@@ -0,0 +1,197 @@
+fi
+
+# Check if already exists
+if [ -d "$SIMULATIONS_DIR/$SIM_NAME" ]; then
+    echo -e "${RED}Error: Simulation '$SIM_NAME' already exists${NC}"
+    exit 1
+fi
+
+# Select template
+echo ""
+echo "Available templates:"
+templates=($(find "$SIMULATIONS_DIR" -maxdepth 1 -name "template_*" -type d -exec basename {} \; | sort))
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)